### ISSUE TITLE
Dependabot doesn't correctly handle patch updates if compatible version specifier (~=) is used in pip

### ISSUE THREAD

**Issue by ankitvgupta** [dependabot/dependabot-core#4989]
I'm a dependabot user for pip packages in a private repo. I made a public repo to reproduce an error with how dependabot handles patch versions. Specifically, I have this block in my dependabot.yml:

``` yaml
    ignore:
      # ignore all patch updates 
      - dependency-name: "*"
        update-types: ["version-update:semver-patch"]
```

This should mean that all pip semvar patch updates get skipped by dependabot. However, dependabot doesn't seem to follow this when using the pip "compatible versioning" syntax. [See PEP 440](https://peps.python.org/pep-0440/#compatible-release). Specifically, when my requirements file looks like this: 

``` yaml
absl-py ~= 1.0.0
aiohttp ~= 3.7.0
bokeh ~= 2.4.0
boto3 ~= 1.21.0
botocore ~= 1.24.0
```
as opposed to the pinned release version format like this: 
``` yaml
absl-py == 1.0.0
aiohttp == 3.7.0
bokeh == 2.4.0
boto3 == 1.21.0
botocore == 1.24.0
```

To recreate the error, I made a [public git repo here ](https://github.com/ankitvgupta/dependabot-demo) with both types of requirements files, and added dependabot to it. It should have been that *just* aiohttp was updated, since that was the only one with a minor release and not just patch releases. This worked for the pinned release version, but not for the compatible release version. 

I also pushed the dependabot logs for the [pinned release version that works correctly ](https://github.com/ankitvgupta/dependabot-demo/blob/main/pinned_release_logs.txt) as well as the [compatible release version which does not](https://github.com/ankitvgupta/dependabot-demo/blob/main/compatible_release_logs.txt). This block at the end of the latter seems problematic: 
<img width="644" alt="image" src="https://user-images.githubusercontent.com/5473865/163076628-98aafe4f-2841-4ed3-9fa0-f5610976fd70.png">
Note the lack of "from" version. 

Any suggestions on how to fix this?

**Package ecosystem**
pip
**Package manager version**
n/a
**Language version**
n/a
**Manifest location and content prior to update**

**dependabot.yml content**
[
](https://github.com/ankitvgupta/dependabot-demo/blob/main/.github/dependabot.yml)

**Updated dependency**
N/A
**What you expected to see, versus what you actually saw**
I expected to only have aiohttp be bumped.
**Native package manager behavior**
N/A
**Images of the diff or a link to the PR, issue or logs**
See the open PRs here: https://github.com/ankitvgupta/dependabot-demo/pulls
**🕹 Bonus points: Smallest manifest that reproduces the issue**
Reproduced in this repo: https://github.com/ankitvgupta/dependabot-demo/

**Comment by deivid-rodriguez** [comment]
Hi, sorry for not getting back earlier, we're slowly going through our backlog.

Can you try enabling the `increase-if-necessary` versioning strategy? I think that should fix the issue!

**Comment by deivid-rodriguez** [comment]
Just to clarify, currently `update-types` ignore conditions are only respected for exact requirements or inexact requirements backed by a lock file. I consider this an issue, but not sure when we'll get to it. 

However, we now support the `increase-if-necessary` versioning strategy in Python, and in this particular case it should give you the same result I believe.

**Comment by ankitvgupta** [comment]
Thanks for the suggestion @deivid-rodriguez. 

Could you help clarify how this strategy applies here? The docs say that with the `increase-if-necessary` strategy, it'll open a new PR "only when required by the new version". What exactly does that mean? As in, what is the "new version" that is "requiring" the PR?

**Comment by deivid-rodriguez** [comment]
Of course! I think the docs could be improved to say "only when required by the latest version", I think that's easier to understand.

In your case, if you have compatible version pins like this

```
absl-py ~= 1.0.0
aiohttp ~= 3.7.0
bokeh ~= 2.4.0
boto3 ~= 1.21.0
botocore ~= 1.24.0
```

and the latest versions are `absl-py-1.0.0`, `aiohttp-3.8.1`, `bokeh-2.4.2`, `boto3-1.21.39`, and `botocore-1.24.39`.

Then with the `increase` strategy (the default), you'll get the following updates

```
absl-py ~= 1.0.0
aiohttp ~= 3.8.1
bokeh ~= 2.4.2
boto3 ~= 1.21.39
botocore ~= 1.24.39
```

That's not what you want because you get updates for patch level releases.

However, with the `increase-if-necessary` strategy, you should only get the `aiohttp` line updated to `aiohttp ~= 3.8.1` and no other updates, because that's the only requirement that needs to be changed in order to match all the latest versions of the dependencies.

Does that make sense?

**Comment by kvmware** [comment]
https://github.com/orgs/community/discussions/52182 this i believe is the inverse of the problem seen here
